InterfaceShop.java

```diff
@@ -0,0 +1,6 @@
+import java.util.List;
+
+public interface InterfaceShop {
+    List<Product> getProducts(String category);
+    void purchaseProduct(int id, User user);
```

Comment:
What is the id here? Probably should be productId to be clear what id we are talking about